features/zones/Create.feature

```diff
@@ -109,8 +110,25 @@ Feature: Zones: Zone create flow
             reason: "invalid characters. Valid characters are numbers, lowercase latin letters and '-', '_' symbols."
       """
 
-    And I "type" "15" into the "$name-input" element
+    # Note: We're deliberately using a valid name here in order to not trigger client-side validation.
+    And I "type" "test" into the "$name-input" element
     And I click the "$create-zone-button" element
 
     Then the "$error" element contains "Invalid zone name"
+    And the "$name-input-group" element contains "invalid characters. Valid characters are numbers, lowercase latin letters and '-', '_' symbols."
+    And the "$instructions" element doesn't exist
+
+  Scenario: The form shows expected error for client-side name validation
+    When I "type" "zone.eu" into the "$name-input" element
+    And I click the "$create-zone-button" element
+
+    Then the "$error" element doesn't exist
+    And the "$name-input-group" element contains "invalid characters. Valid characters are numbers, lowercase latin letters and '-', '_' symbols."
```

Comment:
Why? I want this test to fail if the validation error shown is not this one. We currently only have one case so it’s unambiguous, but if there are more, this test won’t suddenly fail just because we added a new scenario (unless the test data happens to provoke that new scenario which can’t be helped).

Reply:
Because when we can we should try not to assert on human text. We do this a little but already, but I'm hoping to get around to taking a pass at removing as many of those in our tests soonish. If its a hassle, I suppose this one will get spotted when I come to take that pass, so up to you.

Reply:
I don’t understand why this is considered a problem. Asserting against the error message is more useful than just against "input shows error". It makes sure you’re testing the right thing and breaks as soon as this is somehow no longer the case giving you a point to consider whether the test failing is expected (and you need to update it) or not (and something else needs updating).

Yes, that does mean you would need to update the test whenever you change the error message, but that’s completely fine. That doesn’t happen often.

The alternative to this is always pairing the error message with a unique code and add a selectable bit to the input so you can assert against the existence of the "input with invalid dns name error" element, but our backend errors lack the unique identification to do that.

Reply:
`data-testid="dns-error"` would work fine? Anyway as I said no biggie, I hope to do a pass of "remove all human text asserting" in our tests pretty soon. Feel free to resolve.

Reply:
I kinda agree on this. The thing is that in principle you should be able to change i18n values without updating tests.

Why? Because @johnharris85 should be able to just update the yaml file and not understand anything else.

Reply:
I’ve changed this to use `data-test-*` attributes in https://github.com/kumahq/kuma-gui/pull/1444/commits/20a58fc22a7282fd1a113dfd5dc311d740efac4f.